src/main/java/org/onedatashare/server/module/box/BoxResource.java

```diff
@@ -0,0 +1,379 @@
+
+    protected BoxResource(BoxSession session, String path, String id) {
+        super(session, path, id);
+    }
+    protected BoxResource(BoxSession session, String path) {
+        super(session, path);
+    }
+
+
+    //tap outgoing
+    //drain incoming
+    public Mono<Stat> stat() {
+
+        return initialize().map(BoxResource::onStat);
+    }
+
+    public Stat onStat() {
+
+        BoxFolder folder = null;
+
+        BoxItem item;
+
+       // folder.getMetadata().getTypeName();
+        //BoxAPIConnection client = new BoxAPIConnection(session.client.getAccessToken());
+        //URL url = new URL();
+        //BoxAPIRequest req = new BoxAPIRequest(session.client, url, "GET");
+        ;
+        if (id == null){
+           // folder = new BoxFolder(session.client, "0");
+            folder = BoxFolder.getRootFolder(session.client);
+            Iterable<BoxItem.Info> children = folder.getChildren();
+            Stat rStat = buildDirStat(children);
+            rStat.dir = true;
+            rStat.file = false;
+            rStat.name = "root";
+            return rStat;
+           // stat.id = folder.getID();
+        }
+        String type = "";
+            try{
+                folder = new BoxFolder(session.client, id);
+                item = folder;
+                type = item.getInfo().getType();
+            }catch(BoxAPIResponseException e){
+                item = new BoxFile(session.client, id);
+                type = item.getInfo().getType();
+
+            }
+
+            if(type.equals("folder")) {
+                Iterable<BoxItem.Info> children = folder.getChildren();
+                Stat stat = buildDirStat(children);
+                stat.dir = true;
+                stat.file = false;
+                stat.name = folder.getInfo().getName();
+                return stat;
+            }
+            else{
+                BoxFile file = new BoxFile(session.client, id);
+                Stat stat = new Stat();
+                stat.dir = false;
+                stat.file = true;
+                stat.name = file.getInfo().getName();
+                stat.id = id;
+
+                BoxFile.Info fileInfo = file.getInfo();
+                stat.size = fileInfo.getSize();
+                stat.time = fileInfo.getContentModifiedAt().getTime() / 1000;
+                BoxSharedLink sharedLink = fileInfo.getSharedLink();
+                if (sharedLink != null) {
+                    stat.link = sharedLink.toString();
+                }
+                EnumSet<BoxFile.Permission> permissions = fileInfo.getPermissions();
+                if (permissions != null) {
+                    stat.perm = permissions.toString();
+                }
+                return stat;
+
+            }
+        }
+
+    public Stat buildDirStat(Iterable<BoxItem.Info> children){
+        Stat stat = new Stat();
+        ArrayList<Stat> contents = new ArrayList<>();
+        for (BoxItem.Info child : children) {
```

Comment:
Can we create a flux from the iterable and process it asynchronously?
https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Flux.html#fromIterable-java.lang.Iterable-

Reply:
I'll need some suggestions on how to implement this after tomorrows meeting with Yifu and Aashish.  So far what I've tried doesn't seem to work.  I'll keep you updated.

Reply:
Yes, it is a good suggestion but in the transfer module, the mono is blocked until transfer stat is obtained. It might not provide any improvements.

Reply:
That blocking call is bad design. I had coded it that way as a workaround for some issue I was facing during folder-transfer development and it eventually remained that way. I will have that removed and added to the reactive event chain once I am back in Aug.

The intention here is to make the best use of the framework and follow a consistent coding standard. If it doesn't seem feasible, avoid it and go ahead. However, do make an attempt to give it a try.